### PR TITLE
fix amqp thread leak in reconnection and e2e tests stablility fix for Android

### PR DESCRIPTION
**Pull request by jasmineymlo**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/DeviceTwinCommon.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
     private static final String TAG_VALUE = "Tag_Value";
     private static final String TAG_VALUE_UPDATE = "Tag_Value_Update";
 
-    // States of SDK
+    // States of SDKgit s
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/DeviceTwinCommon.java]: typo?

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/IotHubServicesCommon.java [hunk 2]
```diff
@@ -91,6 +90,7 @@ public static void sendMessagesExpectingConnectionStatusChangeUpdate(InternalCli
             @Override
             public void execute(IotHubConnectionStatus status, IotHubConnectionStatusChangeReason statusChangeReason, Throwable throwable, Object callbackContext) {
                 actualStatusUpdates.add(status);
+                System.out.println("*****" + status.toString() + " " + statusChangeReason.toString() + "\n");
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/IotHubServicesCommon.java]: Is this necessary?

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/IotHubServicesCommon.java [hunk 3]
```diff
@@ -412,4 +413,24 @@ public static void waitForStabilizedConnection(List actualStatusUpdates, long ti
...
+    public static void confirmOpenStablized(List actualStatusUpdates, long timeout) throws InterruptedException
+    {
+        long startTime = System.currentTimeMillis();
+        long timeElapsed;
+
+        int numOfUpdates = 0;
+        if (actualStatusUpdates != null)
+        {
+            while (numOfUpdates == 0 || numOfUpdates != actualStatusUpdates.size() || actualStatusUpdates.get(actualStatusUpdates.size() - 1) != IotHubConnectionStatus.CONNECTED) {
+                numOfUpdates = actualStatusUpdates.size();
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/IotHubServicesCommon.java]: bracing style throughout this method

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/SendMessagesCommon.java [hunk 4]
```diff
@@ -414,6 +414,7 @@ public void delayTests()
         {
             throw new RuntimeException(e);
         }
+        System.out.println("*****=======DONE=======*****\n");
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/SendMessagesCommon.java]: Is this necessary?

Comment (jasmineymlo) [reply iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/iothubservices/SendMessagesCommon.java]: fixed

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java [hunk 5]
```diff
@@ -625,9 +633,13 @@ private void openConnection() throws TransportException
                 break;
             case AMQPS:
             case AMQPS_WS:
+                if (scheduledExecutorService == null)
+                {
+                    scheduledExecutorService = Executors.newScheduledThreadPool(1);
```
Comment (prmathur-microsoft) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java]: PSF

**Pull request merged**